scrc/preprocessors/extractors/spider_specific/section_splitting_functions.py

```diff
@@ -25,13 +27,478 @@ def XX_SPIDER(decision: Union[bs4.BeautifulSoup, str], namespace: dict) -> Optio
+            if ti_available is not False and len(ti_) >= len(keywords.items()):
+
+                roles.append(extraction_using_titles(ti_, cm_end_, candidate, sorted_keywords))
+
+            # if titles are not used or not everybody has a title
+            else:
+                # we may be able to extract people using the composition keyword
+                if cm_[0]:
+
+                    roles.append(extraction_using_composition(cm_, cm_end_, candidate, sorted_keywords))
+
+                else:
+                    message = f"We still do not support extraction without tiles and 'composition' keyword"
+                    raise ValueError(message)
+        return roles
+
+    def remove_special_characters(in_str):
+        """
+
+        @param in_str: Input string
+        @return: output string without the following characters: {",",";","s:","e:","******"}
+        """
+        # for ch in characters_to_replace:
+        in_str = in_str.replace(",", "")
+        in_str = in_str.replace(";", "")
+        # some times roles are female or plural
+        in_str = in_str.replace("s:", "")
+        in_str = in_str.replace("e:", "")
+        # remove sequences of '*'
+        star_RegEx = re.compile(r'\*+')
+        star_ = star_RegEx.search(in_str)
+        if star_ is not None:
+            star_s = star_.span()
+            in_str = in_str[:star_s[0] - 1] + in_str[star_s[1] + 1:]
+        return in_str
+
+    def first_last_name(full_name, detail):
+        """
+
+        @param full_name: A string that potentially contains: first name (or initials) and last name
+        @param detail: a dictionary which contains the details of the judicial people
+        @return: the detail dictionary with two new records: first name (or initials) and last name
+        """
+        dot_RegEx = re.compile(r'\w\.\b')
+        dot_ = dot_RegEx.search(full_name)
+        names = full_name.split(" ")
+        last_name = False
+        # remove all the empty strings or strings of length 1
+        for idx, name in enumerate(names):
+            if name == "" or len(name) == 1:
+                names.pop(idx)
+        for name in names:
+            # is it intials?
+            if (dot_ is not None):
+                detail['initials']: name
+                last_name = True
+            # is it the first or the last name?
+            # In some decisions, we only have the last name
+            elif (len(names) > 1 and last_name == False):
+                detail['first name'] = name
+                last_name = True
+            else:
+                detail['last name'] = name
+        return
+
+    def extract_details(roles):
+        """
+
+        @param roles: a list of tuples with judicial people's 1) role 2) details
+        @return: A dictionary ready to be written in the json file
+        """
+        roles_keys = {'pr': 'president', 'as': 'assesseur', 'gr': 'greffier', 'ju': 'juge', 'jusu': 'juge suppléante'}
```

Comment:
use the court_person and court_role here. If your role is not in there, you can add it there